### ISSUE TITLE
SQLITE_PROTOCOL: locking protocol

### ISSUE THREAD

**Issue by beliot-atos** [louislam/uptime-kuma#1172]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/uptime-kuma/security/policy)

### 📝 Describe your problem

Am deploying uptime-kuma on a kubernetes cluster. Am using a persistent volume with a deployment of uptime kuma. Here is the deployment file and pvc:

```
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: uptime-kuma
  namespace: monitoring
spec:
  replicas: 1
  selector:
    matchLabels:
      app: uptime-kuma
  strategy:
    type: Recreate
  template:
    metadata:
      labels:
        app: uptime-kuma
    spec:
      containers:
      - name: uptime-kuma
        image: louislam/uptime-kuma:1
        env:
        - name: UPTIME_KUMA_PORT
          value: "3001"
        imagePullPolicy: Always
        ports:
        - containerPort: 3001
          protocol: TCP
        volumeMounts:
        - mountPath: /app/data
          name: storage
        securityContext:
          allowPrivilegeEscalation: false
        resources:
          limits:
            memory: "512Mi"
            cpu: "1.0"
          requests:
            memory: "256Mi"
            cpu: "0.1"
      volumes:
      - name: storage
        persistentVolumeClaim:
          claimName: uptime-kuma
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: uptimekuma
  namespace: monitoring
spec:
  accessModes:
  - "ReadWriteOnce"
  resources:
    requests:
      storage: 1Gi
  storageClassName: managed-nfs-storage
---
```

Here is the output of the logs for the pod that spun up:
==> Performing startup jobs and maintenance tasks
==> Starting application with user 0 group 0
Welcome to Uptime Kuma
Node Env: production
Importing Node libraries
Importing 3rd-party libraries
Importing this project modules
Prepare Notification Providers
Version: 1.11.3
Creating express and socket.io instance
Server Type: HTTP
Data Dir: ./data/
Copying Database
Connecting to the Database
SQLite config:
[ { journal_mode: 'wal' } ]
[ { cache_size: -12000 } ]
SQLite Version: 3.36.0
Connected
Trace: [Error: SELECT `value` FROM setting WHERE `key` = 'database_version'  limit 1 - SQLITE_PROTOCOL: locking protocol] {
  errno: 15,
  code: 'SQLITE_PROTOCOL'
}
    at process.<anonymous> (/app/server/server.js:1542:13)
    at process.emit (events.js:400:28)
    at processPromiseRejections (internal/process/promises.js:245:33)
    at processTicksAndRejections (internal/process/task_queues.js:96:32)
If you keep encountering errors, please report to https://github.com/louislam/uptime-kuma/issues

When deployment is done without specifying a persistent volume, uptime-kuma works. Was adding a pvc to persist data for when the pod dies or is restarted. Kindly advise.

### 🐻 Uptime-Kuma Version

1.11.3

### 💻 Operating System and Arch

Ubuntu 16.04.7 LTS  

### 🌐 Browser

Microsoft Edge Version 97.0.1072.55 (Official build) (64-bit)

### 🐋 Docker Version

20.10.7

**Comment by chakflying** [comment]
Looks like you are putting your storage on NFS, which is not supported by SQLite.

**Comment by beliot-atos** [comment]
> 

Thank you for this. Let me switch

**Comment by jananathbanuka** [comment]
Btw, how did you come up with those resource `limits` and `requests` values?

**Comment by jledesma84** [comment]
Hey folks. @jananathbanuka @beliot-atos @chakflying I'm using version 1.23.11. I've deployed the application as a docker container on AWS ECS. I used AWS EFS as an external storage. Is EFS not compatible with sqlite? Thanks for your comments

**Comment by CommanderStorm** [comment]
https://sqlite.org/forum/forumpost/8e84712c9d

That might work or might not work. Unclear, as it is still NFS (though v4 which does not guarantee you to corrupt your disks as v3 does).

=> Plan with the occasional db-corruption issue and take regular backups.
=> Plan with a strongly reduced performance ceiling (running a DB on a network file system is not a good idea) 

Tl;DR: Just use a local file system as we suggest in our installation guide.

**Comment by jledesma84** [comment]
Thanks for your comments @CommanderStorm. 
The issue with using a local file system is that if the container dies and a new one is created, the data in the SQLite file that is on the unhealthy container will be lost because a new SQLite file will be created in the new container, right?

**Comment by CommanderStorm** [comment]
That might depend on your cloud provider, but if you use a normal provider your local filesystem is persistent.

`AWS ECS` does not seem to support local, persistent disks. I would not want to be running anything with lower-latency storage requirements (such as a database) on such a system. 

You might want to consider either AWS EC2 or running your container on a cheap VPS (away from the rest of your infrastructure) like [netcup](https://www.netcup.de/)
